src/status_im2/contexts/chat/lightbox/common.cljs

```diff
@@ -0,0 +1,22 @@
+(ns status-im2.contexts.chat.lightbox.common
+  (:require [react-native.reanimated :as reanimated]))
+
+(def flat-list-ref (atom nil))
```

Comment:
Hey guys, I'm seeing from two angles.

The first one is that in React we never define global vars outside the function to hold refs, we always use `useRef` or `createRef` inside component functions. Given that component functions in CLJS are very similar to React components, I think we should stick to this standard.

The other angle is that in Clojure we need strong justification to use global state, and convenience is not enough. I mean, it's a case by case discussion, but it's just a lot more idiomatic to keep state local or manage it in a controlled fashion (e.g. what re-frame does).

>Just to confirm are you talking about the refs only or all atoms?

In my opinion any mutable, global state should be avoided, including atoms. Our codebase has too many unnecessary mutable vars. It's sort of a tech debt to me.

>Also, I previously had these atoms in the view.cljs file, but were globally defined at the top, and they were passed as props to the other components top-view and bottom-view. Is that okay? Or you mean it has to be defined inside the component?

I understand there's some room for subjectivity here, but I think we should define the atoms/refs closest to where they affect the rendering lifecycle (to avoid wasteful re-renders). And from there, they should be passed down as props to other components. If the state needs to be reused across multiple namespaces, it could be promoted to the app db instead.

Reply:
Thanks for the explanation @ilmotta 

Just one more thing, if an `atom` does not change the state of the component, why not define it at the top of the component?

Reply:
Your question is actually difficult to answer because there are many nuances :wink: 

To keep it short, pure functions are easier to reason about and much easier to test. Each global mutable var increases the likelihood for other functions to access and/or mutate them, thus becoming impure. As the system grows, global mutable state tend to cause well known maintainability issues. I believe there are some good blog posts sort of expanding on your question and related to Clojure frontend development, although I don't have the references now.

There's this great paper that touches on many topics related to your question. It's a long read https://curtclifton.net/papers/MoseleyMarks06a.pdf

> Atoms provide a way to manage shared, synchronous, independent state.
> ...
> Since another thread may have changed the value in the intervening time, it may have to retry, and does so in a spin loop.
>
> -- https://clojure.org/reference/atoms

I often try to remember why atoms were created in the first place (their main intent), and it helps me think twice before using them.

Reply:
Thanks, @ilmotta for the clarification and the references. I have fixed it now :)

Reply:
@ilmotta That is indeed a good paper! I first learned about it in this [this David Nolen talk](https://youtu.be/1VLN57wJAcU?t=1539). Speaking of which, David Nolen is a good reference for FP front-end, of course. We should pin that paper somewhere as principles to remember during development/reviews/etc. or at least a few quotes...

Reply:
> Thanks @ilmotta for the clarification and the references. I have fixed it now :)

You just killed a dragon! Thank you

> I first learned about it in this this David Nolen talk

Never seen that one, I'll see if I can take a look. Right! That paper is gold, so many great insights, relatively easy to read too.

> We should pin that paper somewhere as principles 

:sparkling_heart: :sparkles: